### PR TITLE
Supervisor: Refactor and add block height check before stopping IBD

### PR DESCRIPTION
**Pull request by 0xB10C**
This PR refactors the supervisor. To make it more maintainable and readable the single `bbbsupervisor.go` file is split up into multiple files and modules.

Additionally with 2a6ccb7 a block height sanity check is introduced to only stop the IBD after block 596000 (mined on 22/9/2019) or later. 

In this process the version number is bumped to `0.2`.

### CODE REVIEW

File: tools/bbbsupervisor/prometheus/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	if !gjson.Valid(string(body)) { // check if the response is valid json
+		return "", fmt.Errorf("prometheus request returned invalid JSON: %s", string(body))
+	}
+
+	return string(body), nil
+}
+
+// parsePrometheusResponseAsFloat parses a promethues JSON response and returns a float
```
Comment (TheCharlatan) [inline tools/bbbsupervisor/prometheus/prometheus.go]: Typo s/promthues/prometheus/g

**Comment by 0xB10C** [comment]
fixed typo and rebased on master

File: tools/bbbsupervisor/supervisor/util.go [hunk 2]
```diff
@@ -51,3 +51,36 @@ func (s Supervisor) setBBBConfigValue(argument string, value string) error {
...
+	if blockHeight < float64(minHeight) {
+		return fmt.Errorf("current block height (%d) is lower than the minimal block height (%d)", int(blockHeight), minHeight)
+	}
+	return nil
+}
+
+func (s *Supervisor) disableBaseIBDState() (err error) {
+
+	// Before the IBD state of the Base is disabled the block height is sanity checked.
+	// Disabling the ibd state to early results is c-lightning scanning all blocks, which
```
Comment (Tomasvrba) [inline tools/bbbsupervisor/supervisor/util.go]: typos 
> Disabling the ibd state *too* early results *in* c-lightning scanning all blocks

**Comment by Tomasvrba** [comment]
utACK

Since I'm not really familiar with the work you've been doing on the supervisor, I can't comment too much on this. But since it's mostly refactoring and Roland tested the IBD check works, it looks good to me :)

**Comment by 0xB10C** [comment]
Fixed typos

**Pull request merged**